taichi/codegen/codegen_llvm.cpp

```diff
@@ -1623,8 +1640,15 @@ void CodeGenLLVM::visit(ExternalPtrStmt *stmt) {
       llvm::PointerType::get(tlctx->get_data_type(dt), 0));
 
   auto linear_index = tlctx->get_constant(0);
+  int element_shape_idx = 0;
   for (int i = 0; i < num_indices; i++) {
```

Comment:
Just to confirm. In the future `num_indices` should always be equal to `len(array_shape)`, right? @ailzhang @turbo0628

Reply:
If array_shape excludes element_shape, then yes. I plan to remove element shape from the `indices` vector in a follow-up PR.

Reply:
Yup that's exactly the goal of these refactors.

Reply:
Cool!